cartridge/failover/raft.lua

```diff
@@ -0,0 +1,137 @@
+local membership = require('membership')
+local checks = require('checks')
+local log = require('log')
+local topology = require('cartridge.topology')
+local pool = require('cartridge.pool')
+local argparse = require('cartridge.argparse')
+
+local vars = require('cartridge.vars').new('cartridge.failover')
+local PromoteLeaderError = require('errors').new_class('PromoteLeaderError')
+
+vars:new('leader_uuid')
+vars:new('raft_trigger')
+
+--- Generate appointments according to raft status.
+-- Used in 'raft' failover mode.
+-- @function get_appointments
+local function get_appointments(topology_cfg)
+    checks('table')
+    local replicasets = assert(topology_cfg.replicasets)
+
+    local appointments = {}
+
+    for replicaset_uuid, _ in pairs(replicasets) do
+        local leaders = topology.get_leaders_order(
+            topology_cfg, replicaset_uuid
+        )
+        if replicaset_uuid == vars.replicaset_uuid then
+            local my_leader_id = box.info.election.leader
+            local my_leader = box.info.replication[my_leader_id]
+            if my_leader ~= nil then
+                appointments[replicaset_uuid] = my_leader.uuid
+                goto next_rs
+            end
+        end
+
+        local last_leader
+        local last_term = 0
+        for _, instance_uuid in ipairs(leaders) do
+            local server = topology_cfg.servers[instance_uuid]
+            local member = membership.get_member(server.uri)
+
+            if member ~= nil
+            and member.payload.leader_uuid ~= nil
+            and member.payload.raft_term >= last_term
+            then
+                last_leader = member.payload.leader_uuid
+                last_term = member.payload.raft_term
```

Comment:
Is it possible that current instance will have "last_term"? Should it be handled anyhow?

Reply:
And if not - why not? I guess comment required.